wsgi/prz-recipe-options.rst

```diff
@@ -23,30 +18,131 @@ The default is to listen on 0.0.0.0:8080.
+    We will try to cover at least some of the details in later chapters.
 
 wsgi
 ----
 
+This parameter has two forms.
+It can be either ``on`` or ``off``, signaling whether to use the default WSGI server waitress, or not use a WSGI server but use ZServer instead.
+The latter is only an option for Python 2 since ZServer is not (yet?) available for Python 3.
+
+Alternatively you can specify your own PasteDeploy ``ini`` file with this option.
+No ``wsgi.ini`` configuration will be provided in this case.
+Most likely ``wsgi-ini-template`` will be easier to use however, see the next section.
+
+wsgi-ini-template
+-----------------
+
+The path to a WSGI configuration file template for a PasteDeploy compatible WSGI server.
+We will use this form later in the training.
+
 debug-exceptions
 ----------------
 
 Use WSGI middleware debugging facilities.
+When set to ``on`` this option will disable exception views and thus propagate errors to the WSGI stack.
+This allows you to use specific debugging WSGI middleware like the `werkzeug debugger <https://werkzeug.palletsprojects.com/en/0.15.x/debug/>`_.
+We will cover this in the :doc:`debugging chapter <debugging>`.
 
-http-fast-listen
-----------------
+access-log, z2-log
+------------------
+
+``access-log`` has been introduced as an alias for the ``z2-log`` option to specify the location of the access log file.
+
+access-log-level, z2-log-level
+------------------------------
+
+``access-log-level`` is a new alias for ``z2-log-level`` to specify the log level of the access log file.
+For WSGI the default is ``INFO`` (``WARN`` for ZServer).
+
+Advanced logging for WSGI
+-------------------------
+
+``access-log-handler``, ``access-log-args``, ``access-log-kwargs``, ``event-log-handler``, ``event-log-args``, ``event-log-kwargs`` are WSGI only options that you can use to configure arbitrary Python logging handlers.
+Numerous logging handlers are defined in the `Python standard library <https://docs.python.org/3/library/logging.handlers.html>`_.
+
+Exercise 1
+++++++++++
+
+Configure Plone access logging to a TCP Server using ``logging.handlers.SocketHandler`` from the Python standard library.
+Use a local address.
+
+
+..  admonition:: Solution
+    :class: toggle
+
+    Add the following two lines to your buildout configuration (we use ``options.cfg``) from the ``wsgitraining_buildout`` as a starting point:
+
+    .. code-block:: ini
+        :emphasize-lines: 9-10
+
+        ...
+        [instance]
+        recipe = plone.recipe.zope2instance
+        user = admin:admin
+        zeo-client = on
+        zeo-address = 8100
+        shared-blob = on
+        blob-storage = ${buildout:directory}/var/blobstorage
+        access-log-handler = logging.handlers.SocketHandler
+        access-log-args = ('localhost', 9000)
+        eggs =
+            Plone
+            Pillow
```

Comment:
Btw., Pillow is no longer needed here (Plone >=5.0)

Reply:
Resolved this location but there is more of this in other places. I'll create a different PR for this issue once everything else has been merged to master.